Presentation.Web/Controllers/API/V1/OrganizationRegistrationController.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Web.Http;
+using Core.ApplicationServices.Model.Organizations;
+using Core.ApplicationServices.Organizations;
+using Core.DomainModel.ItContract;
+using Core.DomainModel.Organization;
+using Core.DomainServices;
+using Presentation.Web.Controllers.API.V1.Mapping;
+using Presentation.Web.Infrastructure.Attributes;
+using Presentation.Web.Models.API.V1;
+using Presentation.Web.Models.API.V1.Organizations;
+using Swashbuckle.Swagger.Annotations;
+
+namespace Presentation.Web.Controllers.API.V1
+{
+    [PublicApi]
+    [RoutePrefix("api/v1/organization-registrations")]
+
+    public class OrganizationRegistrationController: BaseApiController
+    {
+        private readonly IOrganizationUnitService _organizationUnitService;
+        private readonly IOrgUnitService _orgUnitService;
+
+        public OrganizationRegistrationController(IOrganizationUnitService organizationUnitService,
+            IOrgUnitService orgUnitService)
+        {
+            _organizationUnitService = organizationUnitService;
+            _orgUnitService = orgUnitService;
+        }
+
+        [HttpGet]
+        [Route("{organizationId}/{unitId}")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public HttpResponseMessage GetUnitRegistrations(int organizationId, int unitId)
+        {
+            return _organizationUnitService.GetOrganizationRegistrations(organizationId, unitId)
+                .Select(ToRegistrationDto)
+                .Match(Ok, FromOperationError);
+        }
+
+        [HttpDelete]
+        [Route("{organizationId}/{unitId}")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public HttpResponseMessage RemoveSelectedUnitRegistrations(int organizationId, int unitId, [FromBody] ChangeOrganizationRegistrationRequestDTO requestDto)
+        {
+            var changeParameters = ToChangeParameters(requestDto);
+            return _organizationUnitService.DeleteSelectedOrganizationRegistrations(organizationId, unitId, changeParameters)
+                .Match(FromOperationError, Ok);
+        }
+
+        [HttpDelete]
+        [Route("unit/{organizationId}/{unitId}")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.BadRequest)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public HttpResponseMessage DeleteUnitWithRegistrations(int organizationId, int unitId)
+        {
+            _orgUnitService.Delete(organizationId, unitId);
+            return Ok();
```

Comment:
How should I handle Delete that is void? (it throws an exception in case of an error)

Reply:
you should change from a void of course.. remove the existing endpoint and use of service method